config/routes.rb

```diff
@@ -1,7 +1,15 @@
 Rails.application.routes.draw do
-  resources :users
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
-  # Defines the root path route ("/")
-  # root "articles#index"
-end
+    devise_for :users
+  
+    devise_scope :user do
+      root 'devise/sessions#new'
+    end
+  
+    resources :users, only: [:index, :show] do 
+      resources :posts, only: [:index, :show, :new, :create, :destroy] do
+        resources :comments, only: [:index, :new, :create, :destroy] 
+        resources :likes, only: [:index, :create, :destroy]
+      end
+    end
```

Comment:
- [ ] As per the requirement, please create API endpoints and they should only receive and return JSON. Right now, all of your routes are returning HTML. You can create an API folder under the controllers' directory and put your controllers there for all API requests that are needed.
![image](https://github.com/EmekaEmmanuel/Blograils/assets/91486791/108f1704-5208-4ebe-a699-e92bfb6b92f3)
![image](https://github.com/EmekaEmmanuel/Blograils/assets/91486791/ba1bd6d6-18b4-40ab-a5e5-2fdae8c591f2)
![image](https://github.com/EmekaEmmanuel/Blograils/assets/91486791/200909ce-8994-46e0-a999-1f0908087d2c)